src/digest/base.cr

```diff
@@ -0,0 +1,83 @@
+require "value_ref"
+require "base64"
+
+module Digest::Base(C)
```

Comment:
I think it's an awful shame to add the feature so that our generic variables can have descriptive names, but not use it. Non-generic type names tend not to end in `Type` for obvious reasons, but if we made it the convention that multi-char generic names ended in `Type` would that make it more readable for you?